DevEdu.API/DevEdu.Business.Tests/TopicServiceTests.cs

```diff
@@ -5,130 +5,128 @@
+            var ExpectedTopicId = 42;
 
-
-            _topicRepoMock.Setup(x => x.AddTopic(topicDto)).Returns(TopicData.ExpectedTopicId);
-            _topicRepoMock.Setup(x => x.AddTagToTopic(TopicData.ExpectedTopicId, It.IsAny<int>()));
-
-            var sut = new TopicService(_topicRepoMock.Object);
+            _topicRepository.Setup(x => x.AddTopic(topicDto)).Returns(ExpectedTopicId);
+            _topicRepository.Setup(x => x.AddTagToTopic(ExpectedTopicId, It.IsAny<int>()));            
 
             //When
-            var actualTopicId = sut.AddTopic(topicDto);
+            var actualTopicId = _sut.AddTopic(topicDto);
 
             //Than
-            Assert.AreEqual(TopicData.ExpectedTopicId, actualTopicId);
-            _topicRepoMock.Verify(x => x.AddTopic(topicDto), Times.Once);
-            _topicRepoMock.Verify(x => x.AddTagToTopic(TopicData.ExpectedTopicId, It.IsAny<int>()), Times.Exactly(topicDto.Tags.Count));
+            Assert.AreEqual(ExpectedTopicId, actualTopicId);
+            _topicRepository.Verify(x => x.AddTopic(topicDto), Times.Once);
+            _topicRepository.Verify(x => x.AddTagToTopic(ExpectedTopicId, It.IsAny<int>()), Times.Exactly(topicDto.Tags.Count));            
         }
         [Test]
         public void DeleteTopic_IntTopicId_DeleteTopic()
         {
             //Given
-            const int topicId = TopicData.TopicId;
-
-            _topicRepoMock.Setup(x => x.DeleteTopic(topicId));
+            var topicDto = TopicData.GetTopicDto();
+            var topicId = 1;            
 
-            var sut = new TopicService(_topicRepoMock.Object);
+            _topicRepository.Setup(x => x.DeleteTopic(topicId));
+            _topicRepository.Setup(x => x.GetTopic(topicId)).Returns(topicDto);
 
             //When
-            sut.DeleteTopic(topicId);
+            _sut.DeleteTopic(topicId);
 
             //Than
-            _topicRepoMock.Verify(x => x.DeleteTopic(topicId), Times.Once);
+            _topicRepository.Verify(x => x.DeleteTopic(topicId), Times.Once);
+            _topicRepository.Verify(x => x.GetTopic(topicId), Times.Once);
         }
         [Test]
         public void GetTopic_IntTopicId_GetTopic()
         {
             //Given
             var topicDto = TopicData.GetTopicDto();
-            const int topicId = TopicData.TopicId;
-
-            _topicRepoMock.Setup(x => x.GetTopic(topicId)).Returns(topicDto);
+            var topicId = 1;
 
-            var sut = new TopicService(_topicRepoMock.Object);
+            _topicRepository.Setup(x => x.GetTopic(topicId)).Returns(topicDto);            
 
             //When
-            var dto = sut.GetTopic(topicId);
+            var dto = _sut.GetTopic(topicId);
 
             //Than
             Assert.AreEqual(topicDto, dto);
-            _topicRepoMock.Verify(x => x.GetTopic(topicId), Times.Once);
+            _topicRepository.Verify(x => x.GetTopic(topicId), Times.Exactly(2));           
```

Comment:
please optimize the service's method
it's not cool to have 2 calls of repo's method to get the topic